pkg/dashboard/apiserver/auth/oidc/middleware.go

```diff
@@ -0,0 +1,59 @@
+// Copyright 2024 Chaos Mesh Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package oidc
+
+import (
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"golang.org/x/oauth2"
+
+	"github.com/chaos-mesh/chaos-mesh/pkg/dashboard/apiserver/utils"
+)
+
+func (s *Service) Middleware(c *gin.Context) {
+	ctx := c.Request.Context()
+
+	s.logger.Info("handling oidc middleware")
+	if c.Request.Header.Get("X-Authorization-Method") != "gcp" {
```

Comment:
This part of the code should be updated. We can use `oidc` as the `X-Authorization-Method`. At the same time, we also need to update the frontend to adapt it.

Reply:
Or we can use `oauth` to unify all OAuth-based authorization.